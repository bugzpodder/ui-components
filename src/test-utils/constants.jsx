// @flow
import React from "react";
import { EDC, PIPELINE } from "@grail/lib";

export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const GRAIL = "https://grail.com/wp-content/uploads/2016/12/cropped-final-favicon-32x32.png";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  { word: "Pneumonoultramicroscopicsilicovolcanoconiosis", origin: "General English" },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 1000; i++) {
  // Create about 17,000 records.
  LARGE_EXAMPLE_TABLE_DATA.push(...EXAMPLE_TABLE_DATA);
}

export const EXTRA_LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 100; i++) {
  // Create about 1,700,000 records
  EXTRA_LARGE_EXAMPLE_TABLE_DATA.push(...LARGE_EXAMPLE_TABLE_DATA);
}

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: <div>Over 2500 rounds had been fired with the media claiming that only a few emus had been killed.</div>,
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: <div>Sir George Pearce wages war on the local population of emus in Australia.</div>,
  },
];

export const ALPHABET_ANIMALS_DATA: Array<Array<?string>> = [
  ["alligator", "bat", "cat", "dog", "elephant", "fox"],
  ["goat", "horse", "iguana", "jaguar", "kangaroo", "lion"],
  ["monkey", "narwhal", "octopus", "penguin", "quail", "raccoon"],
  ["seal", "tiger", "umbrella bird", "vulture", "whale", "x-ray tetra"],
  ["yak", "zebra", null, null, null, null],
];

export const TEST_EXTERNAL_DOMAINS: Map<string, string> = new Map().set(EDC, "edc.com").set(PIPELINE, "pipeline.com");
