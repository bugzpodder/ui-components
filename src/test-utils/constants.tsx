import React from "react";

export const COUNTRY_NAMES = [
  "Afghanistan",
  "Akrotiri",
  "Albania",
  "Algeria",
  "American Samoa",
  "Andorra",
  "Angola",
  "Anguilla",
  "Antarctica",
  "Antigua and Barbuda",
  "Argentina",
  "Armenia",
  "Aruba",
  "Ashmore and Cartier Islands",
  "Australia",
  "Austria",
  "Azerbaijan",
  "Bahamas, The",
  "Bahrain",
  "Bangladesh",
  "Barbados",
  "Bassas da India",
  "Belarus",
  "Belgium",
  "Belize",
  "Benin",
  "Bermuda",
  "Bhutan",
  "Bolivia",
  "Bosnia and Herzegovina",
  "Botswana",
  "Bouvet Island",
  "Brazil",
  "British Indian Ocean Territory",
  "British Virgin Islands",
  "Brunei",
  "Bulgaria",
  "Burkina Faso",
  "Burma",
  "Burundi",
  "Cambodia",
  "Cameroon",
  "Canada",
  "Cape Verde",
  "Cayman Islands",
  "Central African Republic",
  "Chad",
  "Chile",
  "China",
  "Christmas Island",
  "Clipperton Island",
  "Cocos (Keeling) Islands",
  "Colombia",
  "Comoros",
  "Congo, Democratic Republic of the",
  "Congo, Republic of the",
  "Cook Islands",
  "Coral Sea Islands",
  "Costa Rica",
  "Cote d'Ivoire",
  "Croatia",
  "Cuba",
  "Cyprus",
  "Czech Republic",
  "Denmark",
  "Dhekelia",
  "Djibouti",
  "Dominica",
  "Dominican Republic",
  "Ecuador",
  "Egypt",
  "El Salvador",
  "Equatorial Guinea",
  "Eritrea",
  "Estonia",
  "Ethiopia",
  "Europa Island",
  "Falkland Islands (Islas Malvinas)",
  "Faroe Islands",
  "Fiji",
  "Finland",
  "France",
  "French Guiana",
  "French Polynesia",
  "French Southern and Antarctic Lands",
  "Gabon",
  "Gambia, The",
  "Gaza Strip",
  "Georgia",
  "Germany",
  "Ghana",
  "Gibraltar",
  "Glorioso Islands",
  "Greece",
  "Greenland",
  "Grenada",
  "Guadeloupe",
  "Guam",
  "Guatemala",
  "Guernsey",
  "Guinea",
  "Guinea-Bissau",
  "Guyana",
  "Haiti",
  "Heard Island and McDonald Islands",
  "Holy See (Vatican City)",
  "Honduras",
  "Hong Kong",
  "Hungary",
  "Iceland",
  "India",
  "Indonesia",
  "Iran",
  "Iraq",
  "Ireland",
  "Isle of Man",
  "Israel",
  "Italy",
  "Jamaica",
  "Jan Mayen",
  "Japan",
  "Jersey",
  "Jordan",
  "Juan de Nova Island",
  "Kazakhstan",
  "Kenya",
  "Kiribati",
  "Korea, North",
  "Korea, South",
  "Kuwait",
  "Kyrgyzstan",
  "Laos",
  "Latvia",
  "Lebanon",
  "Lesotho",
  "Liberia",
  "Libya",
  "Liechtenstein",
  "Lithuania",
  "Luxembourg",
  "Macau",
  "Macedonia",
  "Madagascar",
  "Malawi",
  "Malaysia",
  "Maldives",
  "Mali",
  "Malta",
  "Marshall Islands",
  "Martinique",
  "Mauritania",
  "Mauritius",
  "Mayotte",
  "Mexico",
  "Micronesia, Federated States of",
  "Moldova",
  "Monaco",
  "Mongolia",
  "Montserrat",
  "Morocco",
  "Mozambique",
  "Namibia",
  "Nauru",
  "Navassa Island",
  "Nepal",
  "Netherlands",
  "Netherlands Antilles",
  "New Caledonia",
  "New Zealand",
  "Nicaragua",
  "Niger",
  "Nigeria",
  "Niue",
  "Norfolk Island",
  "Northern Mariana Islands",
  "Norway",
  "Oman",
  "Pakistan",
  "Palau",
  "Panama",
  "Papua New Guinea",
  "Paracel Islands",
  "Paraguay",
  "Peru",
  "Philippines",
  "Pitcairn Islands",
  "Poland",
  "Portugal",
  "Puerto Rico",
  "Qatar",
  "Reunion",
  "Romania",
  "Russia",
  "Rwanda",
  "Saint Helena",
  "Saint Kitts and Nevis",
  "Saint Lucia",
  "Saint Pierre and Miquelon",
  "Saint Vincent and the Grenadines",
  "Samoa",
  "San Marino",
  "Sao Tome and Principe",
  "Saudi Arabia",
  "Senegal",
  "Serbia and Montenegro",
  "Seychelles",
  "Sierra Leone",
  "Singapore",
  "Slovakia",
  "Slovenia",
  "Solomon Islands",
  "Somalia",
  "South Africa",
  "South Georgia and the South Sandwich Islands",
  "Spain",
  "Spratly Islands",
  "Sri Lanka",
  "Sudan",
  "Suriname",
  "Svalbard",
  "Swaziland",
  "Sweden",
  "Switzerland",
  "Syria",
  "Taiwan",
  "Tajikistan",
  "Tanzania",
  "Thailand",
  "Timor-Leste",
  "Togo",
  "Tokelau",
  "Tonga",
  "Trinidad and Tobago",
  "Tromelin Island",
  "Tunisia",
  "Turkey",
  "Turkmenistan",
  "Turks and Caicos Islands",
  "Tuvalu",
  "Uganda",
  "Ukraine",
  "United Arab Emirates",
  "United Kingdom",
  "United States #1",
  "Uruguay",
  "Uzbekistan",
  "Vanuatu",
  "Venezuela",
  "Vietnam",
  "Virgin Islands",
  "Wake Island",
  "Wallis and Futuna",
  "West Bank",
  "Western Sahara",
  "Yemen",
  "Zambia",
  "Zimbabwe",
];

export const COUNTRIES: Array<{
  label: string;
  value: string;
}> = COUNTRY_NAMES.map(name => ({ label: name })).map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const GRAIL =
  "https://grail.com/wp-content/uploads/2016/12/cropped-final-favicon-32x32.png";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  {
    word: "Pneumonoultramicroscopicsilicovolcanoconiosis",
    origin: "General English",
  },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 1000; i++) {
  // Create about 17,000 records.
  LARGE_EXAMPLE_TABLE_DATA.push(...EXAMPLE_TABLE_DATA);
}

export const EXTRA_LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 100; i++) {
  // Create about 1,700,000 records
  EXTRA_LARGE_EXAMPLE_TABLE_DATA.push(...LARGE_EXAMPLE_TABLE_DATA);
}

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct
        trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: (
      <div>
        Over 2500 rounds had been fired with the media claiming that only a few
        emus had been killed.
      </div>
    ),
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: (
      <div>
        Sir George Pearce wages war on the local population of emus in
        Australia.
      </div>
    ),
  },
];

export const ALPHABET_ANIMALS_DATA: Array<Array<string | null>> = [
  ["alligator", "bat", "cat", "dog", "elephant", "fox"],
  ["goat", "horse", "iguana", "jaguar", "kangaroo", "lion"],
  ["monkey", "narwhal", "octopus", "penguin", "quail", "raccoon"],
  ["seal", "tiger", "umbrella bird", "vulture", "whale", "x-ray tetra"],
  ["yak", "zebra", null, null, null, null],
];

export const TEST_EXTERNAL_DOMAINS: Map<string, string> = new Map()
  .set("EDC", "edc.com")
  .set("pipeline", "pipeline.com");
