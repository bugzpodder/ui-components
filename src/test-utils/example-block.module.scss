.container {
  margin: 1rem 0;

  .header {
    font-family: monospace;
  }

  .text {
    background: #f6f8fa;
    font-size: 1rem;
    padding: .5rem;
  }
}