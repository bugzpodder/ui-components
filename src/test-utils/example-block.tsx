import React from "react";
import styles from "./example-block.module.scss";

type Props = {
  content: any;
  strongHeader: string;
  helperText?: string;
};

export const ExampleBlock = (props: Props) => {
  const { strongHeader, helperText = "", content } = props;
  return (
    <div className={styles.container}>
      <pre data-testid="example" className={styles.text}>
        <strong>{strongHeader}</strong>
        {helperText}= {JSON.stringify(content, null, 2)}
      </pre>
    </div>
  );
};
