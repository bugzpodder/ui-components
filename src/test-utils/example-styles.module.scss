.container {
  .spacing {
    margin-top: 10px;
    margin-right: 10px;
    margin-bottom: 20px;
    margin-left: 10px;
  }
}

.spinnerContainer {
  position: relative;

  .buttonContainer {
    display: inline-block;
      position: relative;
  }

  .spacing {
    margin-top: 10px;
    margin-right: 10px;
    margin-bottom: 20px;
    margin-left: 10px;
  }
}

.avatarContainer {
  .inLine {
    display: inline-block;
  }
}

.selectContainer {
  .commonSelect {
    max-width: 350px;
  }
}

.pagedTable {
  height: 600px;
}
