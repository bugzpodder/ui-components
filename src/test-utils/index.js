// @flow
export {
  ALPHABET_ANIMALS_DATA,
  ADAMS_DATA,
  COUNTRIES,
  EMU_DATA,
  EXTRA_LARGE_EXAMPLE_TABLE_DATA,
  LARGE_EXAMPLE_TABLE_DATA,
  EXAMPLE_TABLE_DATA,
  GRAIL,
} from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
export { StyleWrapper as ExampleWrapper } from "../style-wrapper";
