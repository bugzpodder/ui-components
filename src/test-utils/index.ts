export {
  ALPHABET_ANIMALS_DATA,
  ADAMS_DATA,
  COUNTRIES,
  COUNTRY_NAMES,
  EMU_DATA,
  EXTRA_LARGE_EXAMPLE_TABLE_DATA,
  LARGE_EXAMPLE_TABLE_DATA,
  EXAMPLE_TABLE_DATA,
  GRAIL,
  TEST_EXTERNAL_DOMAINS,
} from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
export { StyleWrapper as ExampleWrapper } from "../style-wrapper";
