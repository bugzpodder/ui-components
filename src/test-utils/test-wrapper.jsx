// @flow
import React, { type Node } from "react";
// $FlowFixMe: there is no createGenerateClassName export in @material-ui/core/styles
import { StyleWrapper } from "../style-wrapper";
import { StylesProvider, createGenerateClassName } from "@material-ui/styles";

type Props = {
  children: Node<*>,
};

// TODO(nsawas): Figure out why this new method of disabling global styles
// isnt exactly working. For now we disable all styles with `disableGeneration` prop below.
const generateClassName = createGenerateClassName({
  disableGlobal: true,
});

class MockResizeObserver {
  observe = () => {};
  disconnect = () => {};
}

window.ResizeObserver = MockResizeObserver;

export const TestWrapper = (props: Props) => {
  const { children } = props;
  return (
    <StylesProvider
      disableGeneration
      generateClassName={generateClassName}
    >
      <StyleWrapper>{children}</StyleWrapper>
    </StylesProvider>
  );
};
