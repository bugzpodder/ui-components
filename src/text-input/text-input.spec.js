// @flow
import React from "react";
import { TestWrapper } from "../test-utils";
import { cleanup, render } from "react-testing-library";

import "jest-dom/extend-expect";
import { TextInput } from ".";

afterEach(cleanup);

test("render text field wrapper", async () => {
  const testValue = "test value";
  const testPlaceholder = "Test Placeholder";
  const { container, getByPlaceholderText } = render(
    <TestWrapper>
      <TextInput
        placeholder={testPlaceholder}
        id="operator"
        onChange={() => {}}
        value={testValue}
      />
    </TestWrapper>,
  );
  expect(getByPlaceholderText(testPlaceholder)).toHaveAttribute("value", testValue);
  expect(container).toMatchSnapshot();
});

test("render text field wrapper with readonly as true", async () => {
  const testValue = "test value";
  const { container, getByTestId } = render(
    <TestWrapper>
      <TextInput
        id="operator"
        readOnly
        onChange={() => {}}
        value={testValue}
      />
    </TestWrapper>,
  );
  expect(getByTestId("readonly-text-field")).toHaveTextContent(testValue);
  expect(container).toMatchSnapshot();
});

test("render text field wrapper with readonly as true showing - for empty value", async () => {
  const testValue = "";
  const { container, getByTestId } = render(
    <TestWrapper>
      <TextInput
        id="operator"
        onChange={() => {}}
        value={testValue}
        readOnly
      />
    </TestWrapper>,
  );
  expect(getByTestId("readonly-text-field")).toHaveTextContent("-");
  expect(container).toMatchSnapshot();
});

test("render text field wrapper with readonly as true with showing empty value", async () => {
  const testValue = "";
  const { container, getByTestId } = render(
    <TestWrapper>
      <TextInput
        id="operator"
        onChange={() => {}}
        value={testValue}
        readOnly
        showEmptyValue
      />
    </TestWrapper>,
  );
  expect(getByTestId("readonly-text-field")).toHaveTextContent("");
  expect(container).toMatchSnapshot();
});

test("render text field wrapper with readonly as true with a ReadOnlyComponent", async () => {
  const testValue = "used custom readonly component";
  const testId = "custom-readonly-text";
  const id = "test-readonly-component-id";
  const { container, getByTestId } = render(
    <TestWrapper>
      <TextInput
        id="operator"
        onChange={() => {}}
        value={testValue}
        readOnly
        ReadOnlyComponent="div"
        readOnlyComponentProps={{
          "data-testid": testId,
          id,
        }}
        showEmptyValue
      />
    </TestWrapper>,
  );
  expect(getByTestId(testId)).toHaveTextContent(testValue);
  expect(getByTestId(testId)).toHaveAttribute("id", id);
  expect(container).toMatchSnapshot();
});
