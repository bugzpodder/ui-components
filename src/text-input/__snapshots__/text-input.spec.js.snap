// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render text field wrapper 1`] = `
<div>
  <div
    class="MuiFormControl-root"
    data-testid="text-field"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="operator"
        placeholder="Test Placeholder"
        type="text"
        value="test value"
      />
    </div>
    
  </div>
</div>
`;

exports[`render text field wrapper with readonly as true 1`] = `
<div>
  <span
    class="readonlyTextField"
    data-testid="readonly-text-field"
  >
    test value
  </span>
</div>
`;

exports[`render text field wrapper with readonly as true showing - for empty value 1`] = `
<div>
  <span
    class="readonlyTextField"
    data-testid="readonly-text-field"
  >
    -
  </span>
</div>
`;

exports[`render text field wrapper with readonly as true with a ReadOnlyComponent 1`] = `
<div>
  <div
    data-testid="custom-readonly-text"
    id="test-readonly-component-id"
  >
    used custom readonly component
  </div>
</div>
`;

exports[`render text field wrapper with readonly as true with showing empty value 1`] = `
<div>
  <span
    class="readonlyTextField"
    data-testid="readonly-text-field"
  >
    
  </span>
</div>
`;
